spec/requests/api/v2/access_request_spec.rb

```diff
@@ -24,6 +24,115 @@
 
   subject { response }
 
+  describe 'GET #list' do
+    let(:list_access_requests_route) do
+      get "/api/v2/access_requests/#{admin_user.id}/list",
+          headers: { 'HTTP_AUTHORIZATION' => credentials }
+    end
+
+    context 'when unauthenticated' do
+      before do
+        list_access_requests_route
+      end
+
+      let(:payload) { { email: 'foo@bar' } }
+
+      it { should have_http_status(:unauthorized) }
+    end
+
+    context 'when unauthorized' do
+      let(:unauthorised_user) { create(:user) }
+      let(:payload) { { email: unauthorised_user.email } }
+      let(:unauthorised_user_route) do
+        get "/api/v2/access_requests/#{unauthorised_user.id}/list",
+            headers: { 'HTTP_AUTHORIZATION' => credentials }
+      end
+
+
+      it "should raise an error" do
+        expect { unauthorised_user_route }.to raise_error Pundit::NotAuthorizedError
+      end
+    end
+
+    context 'when authorised' do
+      let(:first_organisation) { create(:organisation, name: 'First Organisation') }
+      let(:second_organisation) { create(:organisation, name: 'Second Organisation') }
+      let(:first_user) {
+        create(:user,
+               first_name: 'First',
+               last_name: 'User',
+               email: 'first_user@test.com',
+               organisations: [first_organisation])
+      }
+      let(:second_user) {
+        create(:user,
+               first_name: 'Second',
+               last_name: 'User',
+               email: "second_user@test.com",
+               organisations: [second_organisation])
+      }
+      let!(:first_access_request) {
+        create(:access_request,
+               first_name: 'test',
+               last_name: 'user',
+               email_address: first_user.email,
+               requester_email: second_user.email,
+               requester_id: second_user.id,
+               organisation: first_organisation.name,
+               reason: 'Need additional support',
+               request_date_utc: '2019-05-05 00:10:47 UTC',
+               status: 'requested')
+      }
+      let!(:second_access_request) {
+        create(:access_request,
+               first_name: 'test',
+               last_name: 'user',
+               email_address: 'test@user.com',
+               requester_email: first_user.email,
+               requester_id: first_user.id,
+               organisation: second_organisation.name,
+               reason: 'Leaving current role',
+               request_date_utc: '2019-05-05 00:10:48 UTC',
+               status: 'requested')
+      }
+      before do
+        list_access_requests_route
+      end
+
+      it 'JSON has a data section with the correct attributes' do
```

Comment:
there should be a spec (this one or a separate one) that tests the filtering down to `requested` in the controller i.e. you'd seed the DB with an actioned access request and ensure that it's not returned in the response.

Reply:
Added this into the (now) #index req spec test by let!'ing an approved access request prior to hitting the endpoint. It's not being parsed into JSON 👍